prow/spyglass/lenses/coverage/coverage.ts

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import Color from "color";
+import {Coverage, parseCoverage} from 'io_k8s_test_infra/gopherage/cmd/html/static/parser';
+import {inflate} from "pako/lib/inflate";
+
+declare const COVERAGE_FILE: string;
+
+const NO_COVERAGE = Color('#FF0000');
+const FULL_COVERAGE = Color('#00FF00');
+
+// Inspired by https://dl.acm.org/citation.cfm?id=949654
+function renderChildren(parent: Node, coverage: Coverage, horizontal: boolean): void {
+  let offset = 0;
+  for (const child of coverage.children.values()) {
+    const node = document.createElement('div');
+    parent.appendChild(node);
+    const percentage = child.totalStatements / coverage.totalStatements * 100;
+    node.style.position = 'absolute';
+    if (horizontal) {
+      node.style.width = `${percentage}%`;
+      node.style.height = '100%';
+      node.style.top = '0';
+      node.style.left = `${offset}%`;
+    } else {
+      node.style.width = '100%';
+      node.style.height = `${percentage}%`;
+      node.style.top = `${offset}%`;
+      node.style.left = `0`;
+    }
+    offset += percentage;
+    if (child.totalFiles === 1) {
+      node.classList.add('leaf');
+      const [filename, file] = child.files.entries().next().value;
+      node.title = `${filename}: ${(file.coveredStatements / file.totalStatements * 100).toFixed(0)}%`;
+      node.style.backgroundColor = NO_COVERAGE.mix(FULL_COVERAGE, file.coveredStatements / file.totalStatements).hex();
+      // Not having a border looks weird, but if we use a constant colour then some parts of this
+      // end up only having
```

Comment:
nit: Clarify?

Reply:
Sorry; nit: Can you clarify this comment? Not sure if something was left off the end or I'm not parsing it right.

Reply:
I think I forgot to finish typing it.